tests/test_agents/test_apy_estimation_abci.py

```diff
@@ -23,27 +23,26 @@
 from aea.configurations.data_types import PublicId
 
 from tests.fixture_helpers import UseGnosisSafeHardHatNet
-from tests.test_agents.base import BaseTestEnd2EndNormalExecution
+from tests.test_agents.base import BaseTestEnd2EndNormalExecution, RoundChecks
 
 
 ipfs_daemon = pytest.mark.usefixtures("ipfs_daemon")
 
-# round check log messages of the happy path
-EXPECTED_ROUND_LOG_COUNT = {
-    "collect_history": 1,
-    "transform": 1,
-    "preprocess": 1,
-    "randomness": 1,
-    "optimize": 1,
-    # One time for training before testing and one time for training on full data after having the final model.
-    "train": 2,
-    "test": 1,
-    "estimate": 2,
-    "cycle_reset": 2,
-    "collect_batch": 2,
-    "prepare_batch": 2,
-    "update_forecaster": 2,
-}
+HAPPY_PATH = (
+    RoundChecks("collect_history"),
+    RoundChecks("transform"),
+    RoundChecks("preprocess"),
+    RoundChecks("randomness"),
+    RoundChecks("optimize"),
+    RoundChecks("train"),
+    RoundChecks("train", success_event="FULLY_TRAINED"),
+    RoundChecks("test"),
+    RoundChecks("estimate", n_periods=2, success_event="ESTIMATION_CYCLE"),
```

Comment:
Here is an example of how we can use other successful events for the happy path's definition.

Reply:
unrelated to this PR an sich, this sentence confuses me. What do you mean by "train on test data" after optimization?

Reply:
Yes, I did not phrase that well. I meant simply on all the available data.

In the beginning, after optimizing, we split the data, train on the train split, and test on the test split. Then, if the performance is acceptable on the test split, we train on all the available data in order to create the final model. 

Hence, you can see why I incorrectly phrased it before.